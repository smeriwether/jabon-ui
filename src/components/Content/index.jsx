import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Editor from '../../components/Editor';
import Project from './components/Project';
import SmallProject from './components/SmallProject';

const propTypes = {
  content: PropTypes.string,
  project: PropTypes.shape({
    wsdlLink: PropTypes.string,
    header1: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    }),
    header2: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    })
  }),
  onContentChange: PropTypes.func.isRequired,
  onProjectChange: PropTypes.func.isRequired,
  expanded: PropTypes.bool
};

const defaultProps = {
  expanded: true,
  project: {
    wsdlLink: '',
    header1: { key: '', value: '' },
    header2: { key: '', value: '' }
  }
};

const classes = {
  expandedEditor: 'h-65 w-100 fl',
  shrinkedEditor: 'h-100-h50 w-100 fl',
  expandedProject: 'h-35 w-100 fl bb bt b--moon-gray',
  shrinkedProject: 'h50 w-100 fl bb bt b--moon-gray'
};

class Content extends Component {
  constructor(props) {
    super(props);

    this.state = {
      project: {
        wsdlLink: props.project.wsdlLink,
        header1: props.project.header1,
        header2: props.project.header2
      },
      expanded: props.expanded
    };
  }

  toggleExpand = () => {
    this.setState((prevState) => ({
      ...prevState,
      expanded: !prevState.expanded
    }));
  }

  onProjectChange = (project) => {
    this.setState((prevState) => ({
      ...prevState,
      project: Object.assign(prevState.project, project)
    }));
    this.props.onProjectChange(project);
  }

  render() {
    const { content, onContentChange } = this.props;

    return (
      <div className='h-100 w-100 br b--moon-gray'>
        <div className={ this.state.expanded ? classes.expandedEditor : classes.shrinkedEditor }>
          <Editor value={ content } name='main-editor' onChange={ onContentChange } />
        </div>
        {
          this.state.expanded ?
            <div className={ classes.expandedProject }>
              <Project
                project={ this.state.project }
                onExpand={ this.toggleExpand }
                onChange={ this.onProjectChange }
              />
            </div>
          :
            <div className={ classes.shrinkedProject }>
              <SmallProject onExpand={ this.toggleExpand } />
            </div>
        }
      </div>
    );
  }
}

Content.propTypes = propTypes;
Content.defaultProps = defaultProps;

export default Content;
