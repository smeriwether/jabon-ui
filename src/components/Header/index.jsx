import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onSubmit: PropTypes.func.isRequired
};

const styles = {
  header: 'h-100 w-100 bg-white flex justify-between items-center pl3 pr3 bb br b--moon-gray',
  button: 'pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box'
};

const Header = ({ onSubmit }) => {
  return (
    <div className={ styles.header }>
      <div className='near-black'>
        <h2>A Modern SOAP Editor</h2>
      </div>
      <div>
        <button data-role='submit' className={ styles.button } onClick={ onSubmit }>
          Submit
        </button>
      </div>
    </div>
  );
};

Header.propTypes = propTypes;

export default Header;
