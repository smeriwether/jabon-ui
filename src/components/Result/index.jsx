import React from 'react';
import PropTypes from 'prop-types';

import EmptyResult from './components/EmptyResult';
import ResultItem from './components/ResultItem';

const propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]).isRequired,
    succeeded: PropTypes.bool.isRequired,
    runtime: PropTypes.number.isRequired,
    content: PropTypes.string
  }))
};

const Result = ({ results }) => {
  const empty = !results || results.length === 0;

  const emptyView = () => {
    return (
      <div className='h-100 w-100 flex justify-center items-center'>
        <EmptyResult />
      </div>
    );
  };

  const resultsView = () => {
    return results.map((result) => {
      return (
        <div className='h400 w-100 pb3' key={ result.id } >
          <ResultItem result={ result } />
        </div>
      );
    });
  };

  return (
    <div className='h-100 w-100 pa3 bg-light-gray bb br b--moon-gray overflow-scroll'>
      { empty ? emptyView() : resultsView() }
    </div>
  );
};

Result.propTypes = propTypes;

export default Result;
