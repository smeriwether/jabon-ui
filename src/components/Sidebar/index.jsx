import React, { Component } from 'react';

import AccountModal from './components/AccountModal';
import Logo from '../../UI/Logo';
import Account from '../../UI/Account';

class Sidebar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      openAccountModal: false
    };
  }

  toggleAccountModal = () => {
    this.setState((prevState) => ({
      ...prevState,
      openAccountModal: !prevState.openAccountModal
    }));
  }

  render() {
    return (
      <div className='h-100 w-100 pb3 bg-blue flex flex-column justify-between items-center'>
        <Logo fill='#357EDD' />
        <Account fill='white' onClick={ this.toggleAccountModal } />

        <AccountModal open={ this.state.openAccountModal } onClose={ this.toggleAccountModal } />
      </div>
    );
  }
};

export default Sidebar;
