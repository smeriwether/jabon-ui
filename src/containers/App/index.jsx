import React, { Component } from 'react';

import Sidebar from '../../components/Sidebar';
import Header from '../../components/Header';
import Content from '../../components/Content';
import Result from '../../components/Result';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      content: '',
      project: {
        wsdlLink: '',
        header1: { key: '', value: '' },
        header2: { key: '', value: '' }
      },
      results: []
    };
  }

  createResult = ({ id, content, succeeded, runtime }) => {
    this.setState((prevState) => ({
      ...prevState,
      results: [{ content, id, succeeded, runtime }, ...prevState.results]
    }));
  }

  onSubmit = () => {
    const { axiosClient } = this.props;
    const { content, project } = this.state;
    const { wsdlLink, header1, header2 } = project;

    const now = Date.now();

    axiosClient
      .post('/soap_requests', {
        content,
        wsdlLink,
        header1,
        header2
      })
      .then((response) => {
        this.createResult({
          id: response.data.id,
          content: response.data.content,
          succeeded: true,
          runtime: Date.now() - now
        });
      })
      .catch((response) => {
        this.createResult({
          id: now,
          content: response.message,
          succeeded: false,
          runtime: Date.now() - now
        });
      });
  }

  onContentChange = (content) => {
    this.setState((prevState) => ({
      ...prevState,
      content
    }));
  }

  onProjectChange = (project) => {
    this.setState((prevState) => ({
      ...prevState,
      project: Object.assign(prevState.project, project)
    }));
  }

  render() {
    return (
      <div className='h-100 w-100'>
        <div className='h-100 w60 fl'>
          <Sidebar />
        </div>
        <div className='h50 w-100-w60 fl'>
          <Header onSubmit={ this.onSubmit } />
        </div>
        <div className='h-100-h50 w-55-w30 fl'>
          <Content
            content={ this.state.content }
            wsdlLink={ this.state.wsdlLink }
            project={ this.state.project }
            onContentChange={ this.onContentChange }
            onProjectChange={ this.onProjectChange }
          />
        </div>
        <div className='h-100-h50 w-45-w30 fl'>
          <Result results={ this.state.results } />
        </div>
      </div>
    );
  }
};

export default App;
