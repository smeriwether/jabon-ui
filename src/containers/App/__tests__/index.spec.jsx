import App from '../../App';

import Sidebar from '../../../components/Sidebar';
import Header from '../../../components/Header';
import Content from '../../../components/Content';
import Result from '../../../components/Result';

describe('<App />', () => {
  const setup = (overrides) => {
    const props = {
      axiosClient: {},
      ...overrides
    };

    const wrapper = shallow(<App { ...props } />);

    return { wrapper };
  };

  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain a Sidebar, Content, and Result component', () => {
    const { wrapper } = setup();

    expect(wrapper.find(Sidebar).length).toEqual(1);
    expect(wrapper.find(Header).length).toEqual(1);
    expect(wrapper.find(Content).length).toEqual(1);
    expect(wrapper.find(Result).length).toEqual(1);
  });
});
