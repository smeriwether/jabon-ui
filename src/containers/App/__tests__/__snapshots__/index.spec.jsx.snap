// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    axiosClient={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="h-100 w60 fl"
        >
          <Sidebar />
        </div>,
        <div
          className="h50 w-100-w60 fl"
        >
          <Header
            onSubmit={[Function]}
          />
        </div>,
        <div
          className="h-100-h50 w-55-w30 fl"
        >
          <Content
            content=""
            expanded={true}
            onContentChange={[Function]}
            onProjectChange={[Function]}
            project={
              Object {
                "header1": Object {
                  "key": "",
                  "value": "",
                },
                "header2": Object {
                  "key": "",
                  "value": "",
                },
                "wsdlLink": "",
              }
            }
            wsdlLink={undefined}
          />
        </div>,
        <div
          className="h-100-h50 w-45-w30 fl"
        >
          <Result
            results={Array []}
          />
        </div>,
      ],
      "className": "h-100 w-100",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Sidebar />,
          "className": "h-100 w60 fl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header
            onSubmit={[Function]}
          />,
          "className": "h50 w-100-w60 fl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Content
            content=""
            expanded={true}
            onContentChange={[Function]}
            onProjectChange={[Function]}
            project={
              Object {
                "header1": Object {
                  "key": "",
                  "value": "",
                },
                "header2": Object {
                  "key": "",
                  "value": "",
                },
                "wsdlLink": "",
              }
            }
            wsdlLink={undefined}
          />,
          "className": "h-100-h50 w-55-w30 fl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "content": "",
            "expanded": true,
            "onContentChange": [Function],
            "onProjectChange": [Function],
            "project": Object {
              "header1": Object {
                "key": "",
                "value": "",
              },
              "header2": Object {
                "key": "",
                "value": "",
              },
              "wsdlLink": "",
            },
            "wsdlLink": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Result
            results={Array []}
          />,
          "className": "h-100-h50 w-45-w30 fl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "results": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="h-100 w60 fl"
          >
            <Sidebar />
          </div>,
          <div
            className="h50 w-100-w60 fl"
          >
            <Header
              onSubmit={[Function]}
            />
          </div>,
          <div
            className="h-100-h50 w-55-w30 fl"
          >
            <Content
              content=""
              expanded={true}
              onContentChange={[Function]}
              onProjectChange={[Function]}
              project={
                Object {
                  "header1": Object {
                    "key": "",
                    "value": "",
                  },
                  "header2": Object {
                    "key": "",
                    "value": "",
                  },
                  "wsdlLink": "",
                }
              }
              wsdlLink={undefined}
            />
          </div>,
          <div
            className="h-100-h50 w-45-w30 fl"
          >
            <Result
              results={Array []}
            />
          </div>,
        ],
        "className": "h-100 w-100",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Sidebar />,
            "className": "h-100 w60 fl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              onSubmit={[Function]}
            />,
            "className": "h50 w-100-w60 fl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Content
              content=""
              expanded={true}
              onContentChange={[Function]}
              onProjectChange={[Function]}
              project={
                Object {
                  "header1": Object {
                    "key": "",
                    "value": "",
                  },
                  "header2": Object {
                    "key": "",
                    "value": "",
                  },
                  "wsdlLink": "",
                }
              }
              wsdlLink={undefined}
            />,
            "className": "h-100-h50 w-55-w30 fl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": "",
              "expanded": true,
              "onContentChange": [Function],
              "onProjectChange": [Function],
              "project": Object {
                "header1": Object {
                  "key": "",
                  "value": "",
                },
                "header2": Object {
                  "key": "",
                  "value": "",
                },
                "wsdlLink": "",
              },
              "wsdlLink": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Result
              results={Array []}
            />,
            "className": "h-100-h50 w-45-w30 fl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "results": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
